File: benchmarks/run_iRF_benchmarks.py
import py_irf_benchmarks_utils
import numpy as np
import argparse
import os

# Global variables
PRINT_SEP = ': '
CSV_DELIMETER = ','
MODEL_SPECS_DIR = './specs'
DATA_DIR = './data'
OUTPUT_DIR = './output'

# Validate that user is running in the benchmarks folderc
py_irf_benchmarks_utils.check_path_exists(DATA_DIR)
py_irf_benchmarks_utils.check_path_exists(OUTPUT_DIR)

# Get the model specs parameters file reference
parser = argparse.ArgumentParser()
parser.add_argument("fname_yaml_specs",
                    help="the yaml with specs to input")
args = parser.parse_args()
print('fname_yaml_specs', args.fname_yaml_specs, sep=PRINT_SEP)

# load specs
inp_fpath_yaml_specs = os.path.join(MODEL_SPECS_DIR,
                                    args.fname_yaml_specs + '.yaml')
py_irf_benchmarks_utils.check_path_exists(inp_fpath_yaml_specs)
print('inp_fpath_yaml_specs', inp_fpath_yaml_specs, sep=PRINT_SEP)

# Read the model specs into a Python dictionary
inp_specs = py_irf_benchmarks_utils.yaml_to_dict(inp_yaml=inp_fpath_yaml_specs)

# Get the raw data name from the model specs
# We will use this to name the output results yaml file later
inp_dsname = inp_specs['inp_dsname'][0]
print('inp_dsname', inp_dsname, sep=PRINT_SEP)

# Get the raw features and responses data
features_csv = os.path.join(DATA_DIR, inp_dsname + "_features.csv")
responses_csv = os.path.join(DATA_DIR, inp_dsname + "_responses.csv")
py_irf_benchmarks_utils.check_path_exists(features_csv)
py_irf_benchmarks_utils.check_path_exists(responses_csv)
print('features_csv', features_csv, sep=PRINT_SEP)
print('responses_csv', responses_csv, sep=PRINT_SEP)

# load data
features = np.loadtxt(features_csv, delimiter=CSV_DELIMETER)
responses = np.loadtxt(responses_csv, delimiter=CSV_DELIMETER)
#print(features[0:10])
#print(responses[0:10])

# output name of the dataset
output_name = args.fname_yaml_specs + "_out.yaml"
print(output_name)

rf_bm = py_irf_benchmarks_utils.consolidate_bm_iRF(features, responses, inp_specs, \
                seed_classifier = 2001, seed_data_split = 24)

py_irf_benchmarks_utils.dict_to_yaml(inp_dict=rf_bm,
                               out_yaml_dir= OUTPUT_DIR,
                               out_yaml_name=output_name)

File: benchmarks/run_iRF_benchmarks.R
#!/usr/bin/env Rscript

library(yaml)
source('R_irf_benchmarks_utils.R')

# Global variables
PRINT_SEP = ': '
CSV_DELIMETER = ','
MODEL_SPECS_DIR = './specs'
DATA_DIR = './data'
OUTPUT_DIR = './output'

if(!dir.exists(DATA_DIR) | !dir.exists(OUTPUT_DIR)){
  stop('Please check you are running this from the benchmarks directory')
  }

# input name of yaml file with model specs
specs_name <- commandArgs(trailingOnly = TRUE)

# input specs
# yaml.load_file('./specs/iRF_mod01.yaml')
spec_path <- paste(MODEL_SPECS_DIR, '/', specs_name, '.yaml', sep = '')
specs <- yaml.load_file(spec_path)
print(spec_path)
#print(specs)

# Get the raw features and responses data
inp_dsname <- specs$inp_dsname
features_path <- paste(DATA_DIR, '/', inp_dsname, "_features.csv", sep = '')
responses_path <- paste(DATA_DIR, '/', inp_dsname, "_responses.csv", sep = '')
print(features_path)
print(responses_path)

# load data
features <- read.csv(features_path, header = FALSE)
responses <- read.csv(responses_path, header = FALSE)
features <- as.matrix(features)
responses <- as.factor(responses$V1)

# make columns 0-indexed to match with python
colnames(features) <- paste('X', 1:dim(features)[2] - 1, sep = '')

# output name of the dataset
irf_bm <- consolidate_bm_iRF(features, responses, specs,
                seed_classifier = 2001, seed_data_split = 24)

# save output
# output name of the dataset
output_path <- paste(OUTPUT_DIR, '/', specs_name, "_out.RData", sep = '')
print(output_path)
save(irf_bm, file=output_path)
